app/api/db/models/user_models.py

```diff
@@ -3,45 +3,44 @@
 from typing import Optional
 from uuid import UUID
 
-from sqlalchemy import Boolean, Column, Date, ForeignKey, Text
-from sqlalchemy.dialects.postgresql import UUID as postgresUUID
+from sqlalchemy import Boolean, Column, Date, Enum, ForeignKey, Text
+from sqlalchemy.dialects import postgresql
 from sqlalchemy.orm import Mapped, relationship
 
 import api.logging
 
-from .base import Base, TimestampMixin, uuid_gen
+from . import base
 
 logger = api.logging.get_logger(__name__)
 
 
-class RoleEnum(str, enum.Enum):
-    USER = "User"
-    ADMIN = "Admin"
-    THIRD_PARTY = "Third Party"
+class RoleType(str, enum.Enum):
+    USER = "USER"
+    ADMIN = "ADMIN"
 
 
-class User(Base, TimestampMixin):
+class User(base.Base, base.IdMixin, base.TimestampMixin):
     __tablename__ = "user"
 
-    user_id: Mapped[UUID] = Column(postgresUUID(as_uuid=True), primary_key=True, default=uuid_gen)
```

Comment:
Just my subjective preference, I like the `user_id` format as you can do `... JOIN USING (user_id)` or sometimes `NATURAL JOIN` when both tables have a `user_id`. But not sure if that would be possible with `IdMixin`.

Reply:
That makes sense, but I feel that since the ORM will reduce the amount of SQL we'd be writing the value of that seems less than the value of what we'd get by standardizing on a primary key name, and there are also situations where the foreign key doesn't use the table name as the prefix. For example if there are two addresses `mailing_address` and `residential_address` and they both have foreign keys into the `address` table, it'd be `mailing_address_id` and `residential_address_id`, not `address_id`. I think it's still possible with `IdMixin` but it gets more complicated. I think it's possible to write an IdMixin that adds a different id name.

Reply:
Thinking more, i think the main argument is that most of the code we read/write will be in the application code not in SQL, so 99% of the code would be things like `user.id` vs `user.user_id` rather than what the where clause would be.

Reply:
There is an alternative that might satisfy both scenarios. If you define a column as:
`id: uuid = Column("user_id", UUID(), primary_key=True)`

Then in the DB itself, it'll create a column called `user_id`, but you use it in the code as `id`. Basically an alias. We never used that in the Mass project, but on NJ UI have used it extensively for columns that are very confusingly named. This however wouldn't let you use the mixin approach.